src/main/java/logging/Person.java

```diff
@@ -0,0 +1,109 @@
+package logging;
+
+import org.apache.log4j.Logger;
+import org.apache.log4j.PropertyConfigurator;
+
+public class Person {
+    private String name;
+    private String surname;
+    private String email;
+    private ChildCount childCount;
+    private Pet pet;
+
+    private static Logger logger;
+
+    enum ChildCount {
+        ONE,
+        TWO,
+        TOO_MANY
+    }
+    enum Pet {
+        CAT,
+        DOG,
+        FISH
+    }
+
+    public Person(String name, String surname, String email, int childcount, String pet) {
+        setupLog4J();
+        this.name = name;
+        this.surname = surname;
+        this.email = email;
+        if (childcount == 1){
```

Comment:
Hard to read, constructor is too big.
Create private methods `int countChildren(int childcount)` and `String whoIsPet(String pet)`.
Also pet =null will throw NPE